src/main/java/com/greedobank/cards/validator/ValueOfEnum.java

```diff
@@ -0,0 +1,20 @@
+package com.greedobank.cards.validator;
+
+import javax.validation.Constraint;
+import javax.validation.Payload;
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Target({ElementType.METHOD, ElementType.FIELD, ElementType.ANNOTATION_TYPE, ElementType.CONSTRUCTOR, ElementType.PARAMETER, ElementType.TYPE_USE})
```

Comment:
I really like this approach, but actually, Spring may handle enums (un)marshalling and (de)serialization with enums in DTO
I would even suggest looking though json schemas

Reply:
I tried different options, but only this one allowed me to get appropriate messages in case smth went wrong